src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -4027,10 +4028,12 @@ private bool IsPossibleParameter(bool allowThisKeyword)
                 case SyntaxKind.OpenBracketToken: // attribute
                 case SyntaxKind.RefKeyword:
                 case SyntaxKind.OutKeyword:
-                case SyntaxKind.OpenParenToken:   // tuple
                 case SyntaxKind.ParamsKeyword:
                 case SyntaxKind.ArgListKeyword:
                     return true;
+                case SyntaxKind.OpenParenToken:   // tuple
+                    return IsFeatureEnabled(MessageID.IDS_FeatureTuples) || IsPredefinedType(this.CurrentToken.Kind);
```

Comment:
If you look at the original change to the parser (see [this diff](https://github.com/dotnet/roslyn/commit/49082d5cd9ff2489d59f9b37e3987f9b98f0ba5e#diff-ac2b4121f27df06f497115cc68277654)), this will make more sense. Basically, I'm restoring the C#6 parser logic when the tuple feature is disabled.